Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -220,22 +220,74 @@ public class PublishPacket: CStruct {
                                 raw_publish_view.user_properties =
                                     UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
                             }
-                            return withOptionalByteCursorPointerFromString(
+                            return withOptionalByteCursorPointerFromStrings(
                             responseTopic,
-                            correlationData,
-                            contentType) { cResponseTopic, cCorrelationData, cContentType in
+                            contentType) { cResponseTopic, cContentType in
                                 raw_publish_view.content_type = cContentType
-                                raw_publish_view.correlation_data = cCorrelationData
                                 raw_publish_view.response_topic = cResponseTopic
 
-                                return body(raw_publish_view)
+                                return withOptionalByteCursorPointerFromData(to: self.correlationData) {  cCorrelationData in
+                                    if let _cCorrelationData = cCorrelationData {
+                                        raw_publish_view.correlation_data = _cCorrelationData
+                                    }
```

Comment:
You don't need the if. We will just assign `raw_publish_view.correlation_data` to nil if cCorrelationData is nil.